pkg/gatherers/conditional/gather_containers_logs.go

```diff
@@ -57,37 +63,29 @@ func (g *Gatherer) gatherContainersLogs(
 	var records []record.Record
 
 	for _, alertLabels := range alertInstances {
-		podNamespace, err := getAlertPodNamespace(alertLabels)
+		info, err := parseAlertLabels(alertLabels)
 		if err != nil {
 			klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
 			errs = append(errs, err)
-			continue
 		}
-		podName, err := getAlertPodName(alertLabels)
-		if err != nil {
-			klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
-			errs = append(errs, err)
+		if len(info.namespace) == 0 {
 			continue
 		}
-		var podContainer string
-		if len(params.Container) > 0 {
-			podContainer = params.Container
-		} else {
-			podContainer, err = getAlertPodContainer(alertLabels)
-			if err != nil {
-				klog.Warningf(logMissingAlert, err.Error(), params.AlertName)
-				errs = append(errs, err)
-			}
-		}
 
 		logContainersFilter := common.LogContainersFilter{
-			Namespace:     podNamespace,
-			FieldSelector: fmt.Sprintf("metadata.name=%s", podName),
+			Namespace: info.namespace,
 		}
 
-		// The container label may not be present for all alerts (e.g., KubePodNotReady).
-		if len(podContainer) > 0 {
-			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", podContainer)
+		if len(params.Container) > 0 {
+			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", params.Container)
+		} else if len(info.container) > 0 {
```

Comment:
`KubePod*` conditions does not have the container.

Reply:
+1 for adding some comment or explanation for this condition, because it looks quite cryptic at the first glance